### PR TITLE
Add documentation about initial_value regarding mock_hw

### LINKED ISSUE #1351: Documentation for mock system's initial_value parameter is missing
We are currently missing documentation for the `initial_value` parameter introduced in #598, at least [not outside of the examples or gazebo](https://control.ros.org/master/search.html?q=initial_value&check_keywords=yes&area=default . I'd be happy to create those, so please feel free to assign me.

### PR DESCRIPTION
**Pull request by fmauch**
This should fix #1351. I added documentation on the mock_hardware page and on the components overview page in the example URDF.

### CODE REVIEW

File: hardware_interface/doc/hardware_interface_types_userdoc.rst [hunk 1]
```diff
@@ -85,7 +85,9 @@ They can be combined together within the different hardware component types (sys
       </gpio>
       <gpio name="flange_analog_IOs">
         <command_interface name="analog_output1"/>
-        <state_interface name="analog_output1"/>    <!-- Needed to know current state of the output -->
+        <state_interface name="analog_output1">    <!-- Needed to know current state of the output -->
+          <param name="initial_value">3.1</param>  <!-- Optional initial value for mock_hardware or gazebo -->
```
Comment (christophfroehlich) [inline hardware_interface/doc/hardware_interface_types_userdoc.rst:89]: does gazebo export gpio interfaces? If not, this comment is misleading here.

**Review by christophfroehlich — APPROVED** [review]
LGTM

File: hardware_interface/doc/hardware_interface_types_userdoc.rst (hunk 1 above)
Comment (fmauch) [reply hardware_interface/doc/hardware_interface_types_userdoc.rst:89]: That's a fair point. I'll change that tomorrow.

File: hardware_interface/doc/mock_components_userdoc.rst [hunk 2]
```diff
@@ -53,3 +53,17 @@ mimic (optional; string)
...
+initial_value (optional; double)
+  Initial value of certain state interface directly after startup. Example:
+
+  .. code-block:: xml
+
+     <state_interface name="position">
+       <param name="initial_value">3.45</param>
+     </state_interface>
+
+  Note: This parameter is shared with the gazebo and gazebo classic plugins.
```
Comment (saikishor) [inline hardware_interface/doc/mock_components_userdoc.rst:69]: may be same as above, mention that the state interface parameters are shared with gazebo and gazebo plugins for interfaces defined under the joint tag

**Review by saikishor — APPROVED** [review]
LGTM

**Review by destogl — APPROVED** [review]

**Review by christophfroehlich — COMMENTED** [review]
Thx!

**Pull request merged**